File: src/src/main/hackerrank/helper/scala/StdIn.scala
package hackerrank.helper.scala

import scala.collection.mutable

object StdIn {
  private var input: mutable.Queue[String] = mutable.Queue()
  private def getNext = if (input.isEmpty) "" else input.dequeue

  def setInputData(in: mutable.Queue[String]) = input = in
  def readInt(): Int = getNext.toInt
  def readLine() = getNext
}

File: src/src/main/java/hackerrank/helper/StdIn.java
package hackerrank.helper;

import java.util.LinkedList;
import java.util.List;

public final class StdIn {

    private LinkedList<String> _input = new LinkedList<>();

    private String getNext() {
        if (_input.isEmpty()) return "";
        else return _input.pop();
    }

    public void setInputData(List<String> in) {
        in.forEach((x) -> _input.add(x));
    }

    public Integer readInt() {
        return Integer.parseInt(this.getNext());
    }

    public String readLine() {
        return this.getNext();
    }
}



